tools/deb/test/machine_setup.sh

```diff
@@ -17,48 +17,163 @@
 
-# Copy config files for DNS
-chmod go+r kubedns
-cp kubedns /etc/dnsmasq.d
-systemctl restart dnsmasq
+  # Cluster settings - the CIDR in particular.
+  cp cluster.env /var/lib/istio/envoy
 
-# Cluster settings - the CIDR in particular.
-cp cluster.env /var/lib/istio/envoy
+  # Copy config files for DNS
+  chmod go+r kubedns
+  cp kubedns /etc/dnsmasq.d
+  systemctl restart dnsmasq
 
-echo "ISTIO_INBOUND_PORTS=80" > /var/lib/istio/envoy/sidecar.env
+  # Update DHCP - if needed
+  grep "^prepend domain-name-servers 127.0.0.1;" /etc/dhcp/dhclient.conf > /dev/null
+  if [[ $? != 0 ]]; then
+    echo 'prepend domain-name-servers 127.0.0.1;' >> /etc/dhcp/dhclient.conf
+    # TODO: find a better way to re-trigger dhclient
+    dhclient -v -1
+  fi
 
-# Update DHCP - if needed
-grep "^prepend domain-name-servers 127.0.0.1;" /etc/dhcp/dhclient.conf > /dev/null
-if [[ $? != 0 ]]; then
-  echo 'prepend domain-name-servers 127.0.0.1;' >> /etc/dhcp/dhclient.conf
-  # TODO: find a better way to re-trigger dhclient
-  dhclient -v -1
-fi
+  chown -R istio-proxy /var/lib/istio/envoy
+}
+
+# Install istio components and certificates. The admin (directly or using tools like ansible)
+# will generate and copy the files and install the packages on each machine.
+function istioInstall() {
+  mkdir -p /etc/certs
+  cp *.pem /etc/certs
+  chown -R istio-proxy /etc/certs
+  # Install istio binaries
+  dpkg -i istio-proxy-envoy.deb
+  dpkg -i istio-agent.deb
+  dpkg -i istio-node-agent.deb
+
+  echo "ISTIO_INBOUND_PORTS=80,9080,3306,27017" > /var/lib/istio/envoy/sidecar.env
+  chown istio-proxy /var/lib/istio/envoy/sidecar.env
+}
+
+function istioRestart() {
+    # Start or restart istio
+    systemctl status istio > /dev/null
+    if [[ $? = 0 ]]; then
+      systemctl restart istio
+    else
+      systemctl start istio
+    fi
+}
+
+
+## Test helpers
+
+# Install a web server and helper packages to help testing and debugging
+# We use a web server to verify HTTP requests can be made from K8S to raw VM and from
+# raw VM to K8S services (zipkin in this test)
+function istioInstallTestHelpers() {
+  NAME=${1-$(hostname)}
 
-# Install istio binaries
-dpkg -i istio-*.deb;
+  sudo apt-get -y install nginx tcpdump netcat tmux
 
-mkdir /var/www/html/$NAME
-echo "VM $NAME" > /var/www/html/$NAME/index.html
+  mkdir /var/www/html/$NAME
+  echo "VM $NAME" > /var/www/html/$NAME/index.html
 
-cat <<EOF > /etc/nginx/conf.d/zipkin.conf
-server {
-      listen 9411;
-      location / {
-        proxy_pass http://zipkin.default.svc.cluster.local:9411/;
-        proxy_http_version 1.1;
+  cat <<EOF > /etc/nginx/conf.d/zipkin.conf
+  server {
+        listen 9411;
+        location / {
+          proxy_pass http://zipkin.default.svc.cluster.local:9411/;
+          proxy_http_version 1.1;
+        }
       }
-    }
 EOF
 
-# Start istio
-systemctl start istio
+
+  systemctl restart nginx
+}
+
+function istioMysql() {
+  # The ratings script is using root/password by default.
```

Comment:
Is this for testing? If so, won't it be easier to just install the docker container and port forward? You won't need any of this. You could probably even use mongo container if you want to and take advantage of continuously tested demo, and simplify future integration tests

Reply:
It may be easier - but not right :-), the goal is to test a real VM. We do have mongodb in docker in 
the cluster. I kept mysql because eventually I want to enable TLS on both mongo and myslq and test it works with the 'bypass auth' or 'custom certs' when they're ready.


I agree it would be good to test the interaction of iptables and envoy running in the raw VM with a docker with forwarded ports, but I think it's lower priority.